airflow/models/dagbag.py

```diff
@@ -306,6 +314,8 @@ def _load_modules_from_file(self, filepath, safe_mode):
         if mod_name in sys.modules:
             del sys.modules[mod_name]
 
+        DagContext.current_autoregister_module_name = mod_name
```

Comment:
This in-place operation feels a bit weird to me… Perhaps a context manager would be easier to understand and maintain? Something like

```python
with DagContext.enable_autoregister(mod_name):
    parse(...)
```

Reply:
Yeah, I agree, This is essentially used as a global variable.

It's not very easy to refactor given it's use across three or so different functions, so a context manager would be tricky. (I.e. I can't see an easy refactor to make it work right now)